wgpu-types: write_only unit tests fail to build with `--no-default-features` (use `std` without the `std` feature)

**Description**

`wgpu-types` is a `#![no_std]` crate that only links the standard library when
the `std` feature is enabled (`#[cfg(feature = "std")] extern crate std;` in
`src/lib.rs`). However, the unit tests in `src/write_only.rs` use
`std::panic::catch_unwind` unconditionally. `catch_unwind` lives in `std` (not
`core`/`alloc`), so building the test target without the `std` feature fails to
resolve `std`.

The `#[cfg(test)] mod tests` block is not gated on the `std` feature, so the
tests are compiled even in configurations where `std` is unavailable

**Repro steps**
From the `wgpu-types` crate:

```sh
cargo test --all-targets --no-default-features
```

**Expected vs observed behavior**
Expected: the tests either build, or are not compiled when `std` is unavailable.

Observed: compilation fails:
```
error[E0433]: failed to resolve: use of unresolved module or unlinked crate std
   --> src/write_only.rs:832:23
    |
832 |         let payload = std::panic::catch_unwind(AssertUnwindSafe(f))
    |                       ^^^ use of unresolved module or unlinked crate std

error[E0282]: type annotations needed
   --> src/write_only.rs:838:40
    |
838 |             if let Some(&string) = payload.downcast_ref::<&'static str>() {
    |                                    ^^^^^^^ cannot infer type

(The `E0282` is cascade fallout from the failed `std` resolution.)
```
The library itself is unaffected — `cargo build --no-default-features` and
`cargo test --no-default-features --features std` both succeed

**Platform**
- wgpu-types 29.0.3
- rustc 1.94.0
- Debian 14

The issue was discovered during its packaging for Debian

--- a/naga-types/src/lib.rs
+++ b/naga-types/src/lib.rs
@@ -1,7 +1,7 @@
 #![cfg_attr(docsrs, feature(doc_cfg))]
 #![no_std]
 extern crate alloc;
-#[cfg(feature = "std")]
+#[cfg(any(feature = "std", test))]
 extern crate std;
 
 pub mod glsl;

--- a/wgpu-core/src/lib.rs
+++ b/wgpu-core/src/lib.rs
@@ -64,7 +64,7 @@
 
 extern crate alloc;
 extern crate naga_types as nt;
-#[cfg(feature = "std")]
+#[cfg(any(feature = "std", test))]
 extern crate std;
 extern crate wgpu_hal as hal;
 extern crate wgpu_types as wgt;

--- a/wgpu-hal/src/lib.rs
+++ b/wgpu-hal/src/lib.rs
@@ -243,7 +243,7 @@ extern crate alloc;
 extern crate naga_types as nt;
 extern crate wgpu_types as wgt;
 // Each of these backends needs `std` in some fashion; usually `std::thread` functions.
-#[cfg(any(dx12, gles_with_std, metal, vulkan))]
+#[cfg(any(dx12, gles_with_std, metal, vulkan, test))]
 #[macro_use]
 extern crate std;
 

--- a/wgpu-types/src/lib.rs
+++ b/wgpu-types/src/lib.rs
@@ -14,7 +14,7 @@
 )]
 #![no_std]
 
-#[cfg(feature = "std")]
+#[cfg(any(feature = "std", test))]
 extern crate std;
 
 extern crate alloc;

